### DISCUSSION TITLE
Document Diagnostics left hand side not wrapping, is there a way?

### DISCUSSION THREAD

**Discussion by benfrain** [ibhagwan/fzf-lua#1374] [Q&A] [answered]
At present, I have LSPs which, when using `diagnostics_document` overflow the area on the left in a single line. Is there any way to have those lines wrapping?

I have tried the following but that is not affecting the left hand side.

```lua
km.set(
  "n",
  "<leader>cd",
  ":lua require'fzf-lua'.diagnostics_document({winopts = {preview={wrap='wrap'}}})<cr>",
  { desc = "Document Diagnostics" }
)
```

**Comment by ibhagwan** [comment]
Fzf ~~0.53~~ 0.54 has a new option to wrap the lines, the default keybind is `ctrl-/`, if you wanna rebind it use `keymap.fzf`.
![IMG_2382](https://github.com/user-attachments/assets/a5df9f8b-0deb-46e0-b64f-fedefd5ebfb8)

**Reply by ibhagwan** [reply]
My bad, wrap is was added in 0.54 not 0.53, so make sure you have the latest fzf:
https://github.com/junegunn/fzf/releases/tag/v0.54.0

**Reply by benfrain** [reply]
Ah, I am on 0.53, that will be the problem!

**Reply by ibhagwan** [reply]
To start with wrap on:
```vim
:FzfLua files fzf_opts.--wrap=true
```

Or permanently:
> Place under `diagnostics` if you want this only for diag
```lua
require'fzf-lua'.setup({
  fzf_opts = { ['--wrap'] = true })
})
```

**Reply by benfrain** [reply]
Sorry, I am unsure how I amend my original keybind to use that?

Should it be like this: 

```lua
km.set(
  "n",
  "<leader>cd",
  ":lua require'fzf-lua'.diagnostics_document({fzf_opts = { ['--wrap'] = true }})<cr>",
  { desc = "Document Diagnostics" }
)
```

?

**Reply by ibhagwan** [reply, marked as answer]
Yes, that’s accurate although you can place it inside the setup ( or lazy `opts =`) under diagnostics to apply for both workspace and document diags.

**Comment by Tahinli** [comment]
@ibhagwan 
Hi, I don't know why buy I can't wrap lines by default.

```lua
return {
    "ibhagwan/fzf-lua",
    dependencies = { "nvim-tree/nvim-web-devicons" },
    opts = {
        ["--wrap"] = true,
    },
    config = function()
        require("fzf-lua").setup({
            diagnostics = {
                multiline = true,
            }
        })
    end
}
```

as you can see they are not wrapped by default :/

![image](https://github.com/user-attachments/assets/4b7575eb-4fd4-4954-9f6c-79b37a7439bc)

**Reply by ibhagwan** [reply]
Enabling `multiline` changes the fzf newline to `\0`, might be an upstream issue with wrap, does the toggle wrap key bind works with `alt-/`?

**Reply by Tahinli** [reply]
> Enabling `multiline` changes the fzf newline to `\0`, might be an upstream issue with wrap, does the toggle wrap key bind works with `alt-/`?

yes alt-/ works fine and with or without multiline I couldn't active wrap by default.

**Reply by ibhagwan** [reply]
> > Enabling `multiline` changes the fzf newline to `\0`, might be an upstream issue with wrap, does the toggle wrap key bind works with `alt-/`?
> 
> yes alt-/ works fine and with or without multiline I couldn't active wrap by default.

Open an issue and I’ll investigate why this happens and whether it’s solved from fzf-lua or  upstream.
